mesonbuild/msubprojects.py

```diff
@@ -376,12 +382,16 @@ def update_git(self) -> bool:
                 success = self.git_rebase(revision)
         else:
             # We are in another branch, either the user created their own branch and
-            # we should rebase it, or revision changed in the wrap file and we need
-            # to checkout the new branch.
+            # we should rebase it, or revision changed in the wrap file (we
+            # know this when the current branch has an upstream) and we need to
+            # checkout the new branch.
             if options.reset:
                 success = self.git_checkout_and_reset(revision)
             else:
-                success = self.git_rebase(revision)
+                if self.git_branch_has_upstream():
+                    success = self.git_checkout_and_rebase(revision)
```

Comment:
I'm not sure about this case, the upstream branch could be in my fork. 

- cd subprojects/myproj
- git checkout -b fix-something
- git push xclaesse fix-something
- cd ../..
- meson subprojects update

I don't want to checkout upstream branch, I want to remain in fix-something but rebased on upstream branch. Maybe we should check if branch exists on `origin` ?

Reply:
Hm, we could match that the remote URL matches what's in the wrap file (either url or push-url). I think that's slightly more accurate.